### PR TITLE
Machine Learning Data selection in DVH and Pyradiomics

### PR DESCRIPTION
**Pull request by Yc325**
it is a new tab in batch preprocessing that allows filtering DVH and Pyradiomics ROIs and then saving modified files.

### CODE REVIEW

File: src/Model/batchprocessing/BatchprocessMachineLearningDataSelection.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        self.dvh_value = dvh_value
+        self.pyrad_value = pyrad_value
+
+        self.dvh_data = None
+        self.pyrad_data = None
+
+        self.full_path_dvh = None
+        self.full_path_pyrad = None
+
+    def start(self):
```
Comment (pedrooste) [inline src/Model/batchprocessing/BatchprocessMachineLearningDataSelection.py:44]: Are you ever checking for the interrupt flag? This is what is called interrupting the processing.

File: src/View/batchprocessing/MachineLearningDataSelectionOptions.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+        self.change_button_dvh_data.setStyleSheet(self.stylesheet)
+
+        # Directory text box for Pyradiomics Data
+        self.directory_input_pyrad = QtWidgets.QLineEdit(
+            "No file selected")
+        self.directory_input_pyrad.setStyleSheet(self.stylesheet)
+        self.directory_input_pyrad.setEnabled(False)
+
+        # Button For Pyradiomics Data to set location
+        self.change_button_pyradiomicsData = QtWidgets.QPushButton(
```
Comment (pedrooste) [inline src/View/batchprocessing/MachineLearningDataSelectionOptions.py:61]: any reason for camelCase?

**Review by pedrooste — CHANGES_REQUESTED** [review]
Can't comment on functionality/testing, but nothing big in terms of code defects!

Approved apart from the small comments

File: src/Model/batchprocessing/BatchprocessMachineLearningDataSelection.py (hunk 1 above)
Comment (Yc325) [reply src/Model/batchprocessing/BatchprocessMachineLearningDataSelection.py:44]: Thanks, I removed it

File: src/View/batchprocessing/MachineLearningDataSelectionOptions.py (hunk 2 above)
Comment (Yc325) [reply src/View/batchprocessing/MachineLearningDataSelectionOptions.py:61]: Thanks, I renamed it

**Pull request merged**